db/schema.rb

```diff
@@ -0,0 +1,61 @@
+# This file is auto-generated from the current state of the database. Instead
+# of editing this file, please use the migrations feature of Active Record to
+# incrementally modify your database, and then regenerate this schema definition.
+#
+# This file is the source Rails uses to define your schema when running `bin/rails
+# db:schema:load`. When creating a new database, `bin/rails db:schema:load` tends to
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema[7.0].define(version: 2022_07_05_090154) do
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "comments", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id", null: false
+    t.bigint "post_id", null: false
+    t.text "text"
+    t.index ["post_id"], name: "index_comments_on_post_id"
+    t.index ["user_id"], name: "index_comments_on_user_id"
+  end
+
+  create_table "likes", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id", null: false
+    t.bigint "post_id", null: false
+    t.index ["post_id"], name: "index_likes_on_post_id"
+    t.index ["user_id"], name: "index_likes_on_user_id"
+  end
+
+  create_table "posts", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id", null: false
+    t.string "title"
+    t.text "text"
+    t.integer "comments_counter"
+    t.integer "likes_counter"
+    t.index ["user_id"], name: "index_posts_on_user_id"
+  end
+
+  create_table "users", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.string "name"
+    t.string "photo"
+    t.text "bio"
+    t.integer "posts_counter"
+  end
+
+  add_foreign_key "comments", "posts"
+  add_foreign_key "comments", "users"
+  add_foreign_key "likes", "posts"
+  add_foreign_key "likes", "users"
+  add_foreign_key "posts", "users"
```

Comment:
- [ ] if you look at [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) you can see that user is called author. So you can access it by `article.author` not `article.user`. To do that you have to slightly change the foreign key: 
```ruby
 create_table :likes do |t|
    t.references :author, foreign_key: { to_table: 'users' } # this will create author_id and a corresponding index
    t.references :post, foreign_key: true # this will create post_id and a corresponding index (but students can also add column and index in separate commands)
  end
```

As you can see, if you want user to be called author you only need to add `t.references :author, foreign_key: { to_table: 'users' }`

Kindly update `user` foreign key in relevant tables (posts, comments, likes)

PS: This can make the app feel complicated and messy for now, but this will surely help you to understand concepts. I had to go back to this project once I started applying for jobs to understand it better 😁

Reply:
Thanks for the review @ShoiraTa  let me check the ERD once more.

Reply:
sure, if you have questions, feel free to ask